src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2605,7 +2608,7 @@ protected override void VisitArguments(ImmutableArray<BoundExpression> arguments
             throw ExceptionUtilities.Unreachable;
         }
 
-        private ImmutableArray<TypeSymbolWithAnnotations> VisitArguments(
+        private ImmutableArray<VisitResult> VisitArguments(
```

Comment:
>ImmutableArray<VisitResult> VisitArguments [](start = 16, length = 42)

:memo: here we visit arguments and store the full result, so that we can use both the result type and the l-value type later on.